bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/ReplicationWorker.java

```diff
@@ -451,6 +455,10 @@ boolean isRunning() {
         return workerRunning && workerThread.isAlive();
     }
 
+    boolean isInReadOnlyMode() {
```

Comment:
Shouldn't the isInReadOnlyMode be a function exposed by the Bookie ? Because this is a bookie-related property and not a ReplicationWorker related property IIRC. The ReplicationWorker could still pipe it through if needed.